src/pages/patientView/genomicOverview/Tracks.tsx

```diff
@@ -5,21 +5,32 @@ import * as tracksHelper from './tracksHelper';
 import {CopyNumberSeg, Mutation, Sample} from 'shared/api/generated/CBioPortalAPI';
 import SampleManager from "../SampleManager";
 import {ClinicalDataBySampleId} from "../../../shared/api/api-types-extended";
+import { IKeyedIconData, IIconData } from './GenomicOverviewUtils';
 
 interface TracksPropTypes {
     mutations:Array<Mutation>;
     cnaSegments:Array<CopyNumberSeg>;
     sampleManager:SampleManager;
     samples:Sample[];
     width:number;
+    mutationGenePanelIconData?:IKeyedIconData;
+    copyNumberGenePanelIconData?:IKeyedIconData;
 }
 
 export const DEFAULT_GENOME_BUILD="GRCh37";
 
+
 export default class Tracks extends React.Component<TracksPropTypes, {}> {
 
     componentDidMount() {
+        this.drawTracks();
+    }
+
+    componentDidUpdate() {
+        this.drawTracks();
```

Comment:
this will result in unnecessary re-render of the visualization because update will fire every time component parents render cycle kicks in whether or not props have changed.   where rendering is non-negligible (here), we should use an optimization like should componentUpdate or pure renderer to prevent this.   We can merge this but lets revisit.

Reply:
@alisman yes, let's discuss this and provide an update.